src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -461,4 +463,24 @@ protected function isGrantedOr403($permission)
             throw new AccessDeniedHttpException();
         }
     }
+
+    protected function isApiRequest()
+    {
+        if (!$this->config->isApiRequest()) {
+            return false;
+        }
+
+        $bundles = array(
+            'JMSSerializerBundle',
+            'BazingaHateoasBundle'
```

Comment:
IIRC Without enable BazingaHateoasBundle for API returns, the returned JSON is OK, but there is never results ...
So seems BazingaHateoasBundle is mandatory for API implementation.